ruma-federation-api/src/event/get_room_state/v1.rs

```diff
@@ -0,0 +1,64 @@
+//! [GET /_matrix/federation/v1/state/{roomId}](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-state-roomid)
+
+use ruma_api::ruma_api;
+use ruma_events::pdu::Pdu;
+use ruma_identifiers::{EventId, RoomId};
+
+ruma_api! {
+    metadata: {
+        description: "Retrieves a snapshot of a room's state at a given event.",
+        method: GET,
+        name: "get_room_state",
+        path: "/_matrix/federation/v1/state/:room_id",
+        rate_limited: false,
+        requires_authentication: true,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    request: {
+        /// The room ID to get state for.
+        #[ruma_api(path)]
+        pub room_id: &'a RoomId,
+
+        /// An event ID in the room to retrieve the state at.
+        #[ruma_api(query)]
+        pub event_id: &'a EventId,
+    }
+
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    response: {
+        /// The full set of authorization events that make up the state of the
+        /// room, and their authorization events, recursively.
+        ///
+        /// Note that events have a different format depending on the room
+        /// version - check the [room version specification] for precise event
+        /// formats.
+        ///
+        /// [room version specification]: https://matrix.org/docs/spec/index.html#room-versions
```

Comment:
Same here, this doesn't need to be stated in our docs since it's obvious from the types (`Pdu` is an enum over different room version's PDUs).

```suggestion
```